src/server.c

```diff
@@ -6151,25 +6175,58 @@ sds genRedisInfoStringCommandStats(sds info, dict *commands) {
     return info;
 }
 
+
 /* Create the string returned by the INFO command. This is decoupled
  * by the INFO command itself as we need to report the same information
  * on memory corruption problems. */
-sds genRedisInfoString(dict *section_dict, int has_all_sections, int has_everything) {
+sds genRedisInfoString(client * c, const char * source) {
+    int default_sections = 0;
+    int all_sections = 0;
+    int everything = 0;
+    int is_sentinel = 0;
+    
+    if (!strcasecmp(source,"sentinel"))
+        is_sentinel = 1;
+
+    dict * section_dict = dictCreate(&stringSetDictType); /* Set to add the subsections to print*/
+    
+    if (!!strcasecmp(source,"sentinel") || !strcasecmp(source,"server")) {
+        if (c == NULL || c->argc == 1) {
+            default_sections = 1;
+        }
+        else {
+            for (int i = 1; i < c->argc; i++) {
+                if (!strcasecmp(c->argv[i]->ptr,"default")) {
+                    default_sections = 1;
+                } else if (!strcasecmp(c->argv[i]->ptr,"all")) {
+                    all_sections = 1;
+                } else if (!strcasecmp(c->argv[i]->ptr,"everything")) {
+                    everything = 1;
+                } else {
+                    sds section = sdsnew(c->argv[i]->ptr);
+                    sdstolower(section);
+                    dictAdd(section_dict,section,NULL);
+                }
+            }
+        }
+    }
```

Comment:
again, why not extract this code to a separate function that returns a dict and two booleans?
then genRedisInfoString will take this input, but it means in between them the sentinel can modify it.